src/components/categories-list/CategoriesList.jsx

```diff
@@ -0,0 +1,94 @@
+import Box from '@mui/material/Box'
+import Button from '@mui/material/Button'
+import Grid from '@mui/material/Grid'
+import CategoryItemCard from '~/components/category-item-card/CategoryItemCard'
+import icons from '~/assets/mui-icons/mui-icons'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { useState, useEffect, useMemo } from 'react'
+import { categoryService } from '~/services/category-service'
+import { styles } from '~/components/categories-list/CategoriesList.styles'
+
+const CategoriesList = () => {
+  const [categories, setCategories] = useState([])
+  const [dataFetched, setDataFetched] = useState(false)
+  const [displayedCategories, setDisplayedCategories] = useState([])
+  const [startIndex, setStartIndex] = useState(0)
+  const categoriesPerPage = 12
+  const colorPairs = useMemo(
+    () => [
+      { backgroundColor: 'rgba(121, 178, 96, 0.2)', color: '#79B260' },
+      { backgroundColor: 'rgba(179, 89, 105, 0.2)', color: '#B35969' },
+      { backgroundColor: 'rgba(241, 188, 25, 0.2)', color: '#F1BC19' },
+      { backgroundColor: 'rgba(0, 167, 167, 0.2)', color: '#00A7A7' },
+      { backgroundColor: 'rgba(90, 128, 136, 0.20', color: '#5A8088' }
+    ],
+    []
+  )
+
+  useEffect(() => {
+    const fetchCategories = async () => {
+      const result = await categoryService.getCategories()
```

Comment:
If I understood correctly, you now fetch all existing categories from the database. Of course, there are not many of them yet. But in the future this may cause performance issues. So, it's better to fetch the data partially. To do this, we have `skip` and `limit` queries.

I don't see such a requirement in the issue, so maybe it's not crucial here. Just keep it in mind for the future.